sdk/typescript/src/rpc/faucet-client.ts

```diff
@@ -23,8 +25,20 @@ export async function requestSuiFromFaucet(
       ...(httpHeaders || {}),
     },
   });
-  const parsed = await res.json();
 
+  if (res.status === 429) {
+    throw new FaucetRateLimitError(
+      `Too many requests from this client have been sent to the faucet. Please retry later`
+    );
+  }
+  let parsed;
+  try {
+    parsed = await res.json();
+  } catch (e) {
+    throw new Error(
+      `Ecountered error when parsing response from faucet, error: ${e}, status ${res.status}, response ${res}`
```

Comment:
This error is often due to cloudflare timeout instead of faucet server itself. Cloudflare returns a html doc instead of a valid json

Reply:
this makes sense but even better if we can explicitly see a TIMEOUT and handle that accordingly.

Reply:
Yeah I am hoping with this logging we can see the status code for timeout and other common cloudflare errors